### ISSUE TITLE
Cookies not removed

### ISSUE THREAD

**Issue by gregBerthelot** [KIProtect/klaro#14]
Hi again

On this [site](https://bit.ly/2kzsHo4) all __utm cookies are not removed if you disable Web Analytics app

```
apps : [
        {
            name : 'analytics',
            default: true,
            onlyOnce: true,
            title : 'Web analytics',
            purposes : ['analytics'],
            cookies : ['__utma', '__utmb', '__utmc', '__utmt', '__utmz'],
            callback : function(consent, app){
                console.log("User consent for app "+app.name+": consent="+consent)
            },
            required : false,
            optOut : false,
        }
]
```

Do I miss something?
Thx
Greg

**Comment by adewes** [comment]
Hit @gregBerthelot , thanks for posting the config and site URL, this makes debugging much easier :)
In your source I see two scripts with `data-name` attributes, named `sharing` and `livechat`. In your app config, you also have an app called `analytics`, for which I couldn't find an equivalent `data-name` in the source though. Please check if you have added the proper tag, because if Klaro won't find a given app it will not be able to manage it.

Let me know if that solves your problem, and thanks for trying Klaro!

**Comment by gregBerthelot** [comment]
@adewes Got it, but adding Google Tag like following does not track users correctly in GA:

```
<script async type="opt-in"  data-src="https://www.googletagmanager.com/gtag/js?id=UA-XXXXXXXX-X" data-name="analytics"></script>
<script type="opt-in"
        data-type="application/javascript"
        data-name="analytics">
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-XXXXXXXX-X');
</script>
```

Adding Google Tag like following properly tracks users correctly in GA. It removes well `_gat_gtag_UA_XXXXXXXX-X` and all `__utmX` cookies but unfortunately do not remove `_ga` and `_gid` cookies:

```
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-XXXXXXXX-X"></script>
<script type="opt-in"
        data-type="application/javascript"
        data-name="analytics">
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-XXXXXXXX-X');
</script>
```

Sorry with those issues
Greg

**Comment by adewes** [comment]
No worries, this was a great test case, thanks for reporting this! The problem was that GA sets the cookies for a different domain or path, and in order to reliably delete these cookies we need to exactly match those. As there is no way to get the domain / path of a cookie in JS (unfortunately) there's no other way than to configure this manually.
I just released v0.1.5 of Klaro, which allows you to pass an array for a given entry in the `cookies` list. This array should contain the regex / string pattern (as before), the path (e.g. `/`) and the domain of the cookie (e.g. `.foo.com`). You can get the domain / path e.g. by looking at the developer tools. Klaro will then reliably delete all your cookies.

**Comment by gregBerthelot** [comment]
Can you confirm following ?

´´´
cookies [
[a, '/', 'foo.com'],
[b, '/', 'bar.com'],
 'c'],
´´´

**Comment by adewes** [comment]
Yes that's correct! Let me know if you encounter any problems. We might soon release our auto-configure tool which will make writing and debugging configurations much easier (hopefully).

**Comment by adewes** [comment]
So the `_ga` and `_gat_...` and `_gi` cookies are on `.[your domain].fr` whereas the Klaro cookie is on `www.[yourdomain].fr` (which is also the domain of the current page). It's very unfortunate (but required for security) that JS cannot read cookie paths and domains, so there's no alternative to manually removing the cookies.

**Comment by gregBerthelot** [comment]
Seems working bette with the latest commit and the ability to add domain and path to cookies. Great job.

Despite this I dunno how I can see me in real time in GA without any cookies set for Google :)

BTW have you seen https://github.com/KIProtect/klaro/issues/5
